### DISCUSSION TITLE
Question regarding safe check of part of message + secondary question regarding embedded language

### DISCUSSION THREAD

**Discussion by ghost** [nextgenhealthcare/connect#4568] [Q&A] [answered]
Hello.

Is there any way I can check for the existence of something like `msg['errors'][1]` safely without using multiple ifs to avoid errors ?
For example in this case if only `msg[errors]` existed the initial check for `msg['errors'][0]` will fail.
I'm thinking with a try but I'd really like to ask before since it's a case I need and I want to get it right.

Also is there some kind of reference I can read to see what's possible in this input field ?
https://share.dmca.gripe/mJgJ05wiGCibMEws.png
Is this Java, Javascript or Rhino ( https://github.com/mozilla/rhino ) ? If one of them what versions ? It doesn't seem like modern Javascript and I've seen some Java.

**Comment by tonygermano** [comment, marked as answer]
The code you write is javascript running on the Rhino engine (which allows it to access java libraries, but you are not writing java code.) Rhino supports all of ES5 and some of ES6 and later. You can check which features are supported here https://mozilla.github.io/rhino/compat/engines.html. The version of Rhino you are using depends on your mirth version.

To your first question a `try` will work well. I use this Code Template:
```js
/**
	Try to run a function that may throw an exception and return the function's return value. Accepts
	either a default value or an error handler function in the event of an exception.

	@param {Function} fn - function to run
	@param {Any} onError - either a default value for when fn throws an exception or a callback
		function that accepts the caught exception as a parameter
	@return {Any} the return value of fn or the return value or onError or onError
*/
function tryit(fn, onError) {
	var result;
	try {
		result = fn();
	}
	catch(e) {
		if (typeof onError == 'function') {
			result = onError(e);
		}
		else {
			result = onError
		}
	}
	return result;
}
```
Then I call it like:
```js
var error = tryIt(() => msg['errors'][1], undefined);
```

**Reply by ghost** [reply]
Thanks, this made a lot of things more clear for me.
Is there any place where I can see the version of Rhino for each Mirth ? I'd be mostly interested in 3.7.0 , 3.5.1 and the latest one.

**Reply by tonygermano** [reply]
Look for the rhino .jar file in the server-lib folder of your mirth installation.
